### PR TITLE
feat(provider): Add burstSMS as an SMS provider

### PR DESCRIPTION
**Pull request by dlmcauslan**
### What change does this PR introduce? 

This adds BurstSMS as a new SMS provider https://burstsms.com/

### Other information (Screenshots)

Burst SMS requires an API Key and API Secret to send SMS messages.
It does not require a phone number - it automatically chooses the number for you.
I have included the from phone number as part of the sms options, because the from phone number can be manually specified on a per message basis, instead of letting burst SMS automatically choosing the correct number.

### CODE REVIEW

File: libs/shared/src/consts/providers/channels/sms.ts [hunk 1]
```diff
@@ -78,4 +79,12 @@ export const smsProviders: IProviderConfig[] = [
     docReference: 'https://www.firetext.co.uk/docs',
     logoFileName: { light: 'firetext.svg', dark: 'firetext.svg' },
   },
+  {
+    id: SmsProviderIdEnum.Burstsms,
```
Comment (p-fernandez) [inline libs/shared/src/consts/providers/channels/sms.ts:83]:
```suggestion
    id: SmsProviderIdEnum.BurstSms,
```

File: providers/burstsms/src/index.ts [hunk 2]
```diff
@@ -0,0 +1 @@
+export * from './lib/burstSms.provider';
```
Comment (p-fernandez) [inline providers/burstsms/src/index.ts:1]:
```suggestion
export * from './lib/burst-sms.provider';
```

File: providers/burstsms/src/lib/burstSms.provider.spec.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import { BurstSmsSmsProvider } from './burstSms.provider';
```
Comment (p-fernandez) [inline providers/burstsms/src/lib/burstSms.provider.spec.ts:1]:
```suggestion
import { BurstSmsProvider } from './burst-sms.provider';
```

File: providers/burstsms/src/lib/burstSms.provider.spec.ts [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+import axios from 'axios';
+import { v4 as uuidv4 } from 'uuid';
+
+test('should trigger burstSms axios request correctly', async () => {
```
Comment (p-fernandez) [inline providers/burstsms/src/lib/burstSms.provider.spec.ts:5]:
```suggestion
test('should trigger Burst SMS axios request correctly', async () => {
```

File: providers/burstsms/src/lib/burstSms.provider.spec.ts [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+  });
+
+  jest.spyOn(axios, 'create').mockImplementation(() => {
+    return {
+      post: fakePost,
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    } as any;
+  });
+
+  const provider = new BurstSmsSmsProvider({ apiKey: '', secretKey: '' });
```
Comment (p-fernandez) [inline providers/burstsms/src/lib/burstSms.provider.spec.ts:17]:
```suggestion
  const provider = new BurstSmsProvider({ apiKey: '', secretKey: '' });
```

File: providers/burstsms/src/lib/burstSms.provider.ts [hunk 6]
```diff
@@ -0,0 +1,48 @@
+import {
+  ChannelTypeEnum,
+  ISendMessageSuccessResponse,
+  ISmsOptions,
+  ISmsProvider,
+} from '@novu/stateless';
+import axios, { AxiosInstance } from 'axios';
+import qs from 'qs';
+
+export class BurstSmsSmsProvider implements ISmsProvider {
```
Comment (p-fernandez) [inline providers/burstsms/src/lib/burstSms.provider.ts:10]:
```suggestion
export class BurstSmsProvider implements ISmsProvider {
```
We do not need to add the type of provider in the name as we can clearly see it from channel property and thus we avoid the cacophony.

File: providers/burstsms/src/lib/burstSms.provider.ts [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+  id = 'burstSms';
```
Comment (p-fernandez) [inline providers/burstsms/src/lib/burstSms.provider.ts:11]:
```suggestion
  id = 'burstsms';
```
Is the value defined in the provider enum of the shared module.
Reference: libs/shared/src/consts/providers/provider.enum.ts

**Review by p-fernandez — CHANGES_REQUESTED** [review]
Thanks for adding this new provider. 🙏🏻 
Mostly naming changes for consistency and few details. Overall good!

File: providers/burstsms/README.md [hunk 8]
```diff
@@ -0,0 +1,9 @@
+# Novu BurstSms Provider
+
+A BurstSms sms provider library for [@novu/node](https://github.com/novuhq/novu)
+
+## Usage
+
+```javascript
+    FILL IN THE INITIALIZATION USAGE
+```
```
[the remark this answers is not in the document]
Comment (dlmcauslan) [reply providers/burstsms/README.md]: I've updated this :)

**Comment by dlmcauslan** [comment]
@p-fernandez I think I've made all your requested changes. Let me know if you're happy with them.
I will merge in `next` once you've taken a look.

File: apps/api/src/app/events/services/sms-service/handlers/burstsms.handler.ts [hunk 9]
```diff
@@ -0,0 +1,18 @@
+import { ChannelTypeEnum } from '@novu/shared';
```
Comment (p-fernandez) [inline apps/api/src/app/events/services/sms-service/handlers/burstsms.handler.ts:1]: I forgot to ask to rename this file to `burst-sms.handler.ts`. Apologies for that! 🙏🏻

File: apps/api/src/app/events/services/sms-service/handlers/index.ts [hunk 10]
```diff
@@ -6,3 +6,4 @@ export * from './sms77.handler';
 export * from './termii.handler';
 export * from './gupshup.handler';
 export * from './firetext.handler';
+export * from './burstsms.handler';
```
Comment (p-fernandez) [inline apps/api/src/app/events/services/sms-service/handlers/index.ts:9]:
```suggestion
export * from './burst-sms.handler';
```
In order to the previous comment. Apologies as I missed it.

File: providers/burst-sms/README.md [hunk 11]
```diff
@@ -0,0 +1,19 @@
...
+
+const provider = new BurstSmsProvider({ 
+  apiKey: process.env.BURST_SMS_API_KEY,        // Your Burst SMS API Key
+  secretKey: process.env.BURST_SMS_SECRET_KEY,  // Your Burst SMS API Secret
+})
+
+await provider.sendMessage({
+  to: '0123456789',
+  content: 'Message to send',
+});
```
Comment (p-fernandez) [inline providers/burst-sms/README.md:18]: 👏🏻

File: providers/burst-sms/src/index.ts [hunk 12]
```diff
@@ -0,0 +1 @@
+export * from './lib/burstSms.provider';
```
Comment (p-fernandez) [inline providers/burst-sms/src/index.ts:1]:
```suggestion
export * from './lib/burst-sms.provider';
```
I missed this in my previous review. Apologies for that.

File: providers/burst-sms/src/lib/burstSms.provider.spec.ts [hunk 13]
```diff
@@ -0,0 +1,31 @@
+import axios from 'axios';
```
Comment (p-fernandez) [inline providers/burst-sms/src/lib/burstSms.provider.spec.ts:1]: I missed to ask for a rename of the test file to: `burst-sms.provider.spec.ts`. Apologies for that.

File: providers/burst-sms/src/lib/burstSms.provider.ts [hunk 14]
```diff
@@ -0,0 +1,48 @@
+import {
```
Comment (p-fernandez) [inline providers/burst-sms/src/lib/burstSms.provider.ts:1]: I missed to ask for a rename of the file to: `burst-sms.provider.ts`. Apologies for that.

**Review by p-fernandez — APPROVED** [review]
I missed to ask for some file renames in my previous review. Apologies for that.
But I am approving before hand as you can do it now before merging `next` so we can see integration checks passing.
Thanks for the contribution. 🌟

File: apps/api/src/app/events/services/sms-service/handlers/burstsms.handler.ts (hunk 9 above)
Comment (dlmcauslan) [reply apps/api/src/app/events/services/sms-service/handlers/burstsms.handler.ts:1]: No worries. Sorry I probably should have caught that one myself when doing the changes yesterday

**Comment by dlmcauslan** [comment]
@p-fernandez I've made those changes, and merged in `next`. Hopefully the PR should be all good now.

**Pull request merged**